### PR TITLE
Fix #1025: Debugger Does Not Enumerate ctypes Arrays

### LINKED ISSUE #1025: Debugger Does Not Enumerate ctypes Arrays
**Describe the bug**
The debugger in Visual Studio 2022 does not treat ctypes arrays as containers, and hovering over them or looking at them in the variables window does not show length or allow the items to be enumerated like a list or dict. This is a new problem in VS2022, and did not occur in VS2017 or VS2019, which is why I've only recently started using VS2022 and this problem is making it very unpleasant as I use ctypes extensively in my work.

Ordinarily I would post in a forum to ask if there is some setting that's set incorrectly (the settings are default for Python), but as there's no forum here I'm making a bug report.

**Steps to Reproduce**
For me this occurs any and all times I use ctypes arrays in any context, so I don't really have any specific instructions to reproduce. The only thing worth noting is that this occurs regardless of whether "Enable debugging of the Python standard library" is enabled (I mostly work with it enabled). Maybe read a random binary file into a bytearray and do something like

x = (ctypes.c_uint8 * len(file_bytes)).from_buffer(file_bytes)

Then hover over x (or look at it in the variables window). It will only show "special variables" and "protected variables".

**Expected behavior**
In past VS versions the debugger would display as if it was a list, showing the length and list of all elements.

**Configuration information**
VS Version: Community 2022 (64-bit) 17.3.0
PTVS version: 17.0.22179.3
Python version: 3.9.7 64-bit (x64)
Debugger Type: I don't think this is relevant for VS2022?

### PR DESCRIPTION
**Pull request by int19h**
Use tuple resolver for ctypes.Array

### CODE REVIEW

File: src/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_xml.py [hunk 1]
```diff
@@ -91,6 +91,12 @@ def _create_default_type_map():
     except:
         pass
 
+    try:
+        from ctypes import Array
```
Comment (fabioz) [inline src/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_xml.py:95]: Seems good to me, but it'd be nice to have a test too...

**Review by fabioz — APPROVED** [review]

**Comment by r3m0t** [comment]
I also have a change for the resolver I want to upstream. Are you accepting PRs or should I go to the pydev.debugger repo?

My change removes useless attributes like x.index when x is a local of type `list`.

**Comment by fabioz** [comment]
@r3m0t

Ideally changes in the debugger implementation would go to the `pydev.debugger` repo initially (because the tests for `pydev.debugger` are only run in that repo, not in this one which is geared towards the integration of `pydevd` with the debug adapter protocol).

Changes should ideally have a test too.

As a note, for the particular feature you're describing, please create an issue first (on the `debugpy` repo is fine) as for your use-case I believe the filtering capabilities should already suffice.

File: src/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_xml.py (hunk 1 above)
Comment (fabioz) [reply src/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_xml.py:95]: I see what you mean (it tests the resolver themselves, not the resolution process), but I think that using `test_resolvers` for that is also a good place (the API to be used is `get_type(obj)` where the last item is the actual resolver... there's only one test there that calls this API, but I think we can extend that same module to add this test).

**Review by fabioz — APPROVED** [review]

**Pull request merged**